Quantum-Control-Applications-QuAM/Superconducting/README.md

```diff
@@ -0,0 +1,357 @@
+The [data handler](https://github.com/qua-platform/py-qua-tools/tree/main/qualang_tools/results#data-handler) is used to save data into an automatically generated folder with folder structure: 
+`{root_data_folder}/%Y-%m-%d/#{idx}_{name}_%H%M%S`.
+
+The saved data can have a different format depending on its type:
+* The figures are saved as .png.
+* The arrays are saved as .npz.
+* The node parameters, state and wiring are saved as .json.
+
+### calibration_graph
+This folder contains all the calibration scripts that can compose a qualibrate graph.
+The structure of the nodes is described below.
+
+### configuration
+The configuration folder contains the python scripts used to build the QUAM before starting the experiments.
+It contains three files whose working principles are explained in more details below:
+* [make_wiring.py](./configuration/make_wiring.py): create the port mapping between the control hardware (OPX+, Octave, OPX1000 LF fem, MW fem) and the quantum elements (qubits, resonators, flux lines...). 
+* [make_quam.py](./configuration/make_quam.py): create the state of the system based on the generated wiring and QUAM components and containing all the information necessary to calibrate the chip and run experiments. This state is used to generate the OPX configuration. 
+* [modify_quam.py](./configuration/modify_quam.py): update the parameters of the state programmatically based on defaults values (previous calibration, chip manufacturer specification...).
+
+### quam_libs
+This folder contains all the utility functions necessary to create the wiring or build the QUAM, as well as QUA macros and data processing tools:
+* [components](./quam_libs/components): this is where the QUAM root and custom QUAM components are defined. A set of basic QUAM components are already present, but advanced user can easily modify them or create new ones.
+* [lib](./quam_libs/lib): contains several utility functions for saving, fitting and post-processing data.
+* [quam_builder](./quam_libs/quam_builder): contains the main functions called in [machine.py](./quam_libs/quam_builder/machine.py) and used to generate the wiring and build the QUAM structure from it and the QUAM components declared in the [components](./quam_libs/components) folder. It also contains the [pulses.py](./quam_libs/quam_builder/pulses.py) file where the default qubits pulses are defined.
+
+## How to generate the QuAM
+Before starting to run experiments, it is necessary to build the Quantum Abstract Machine ([QuAM](https://github.com/qua-platform/quam)) for the desired 
+qubit chip architecture. More details about QUAM itself can be found in the [QuAM documentation](https://qua-platform.github.io/quam/).
+
+The process can be divided into three sections:
+1. First one needs to create the wiring specifying which quantum element (qubit, resonator, flux line...) is connected to 
+which OPX channel. 
+2. Then, the QuAM components can be derived from the resulting wiring.json file. 
```

Comment:
@deanpoulos Yonatan Cohen made a comment that because make_wiring_xxx.py is already declaring the quantum elements of the chip that the user will be working with, asked if we can take a pause and reconsider if a make_wiring_xxx.py would be replaced by make_chip_layout_xxx.py? even though the output of it is to generate wiring.json indeed. But the quam will be ruled to work with the qubits declared by make_wiring

Reply:
I understand Yonatan's comment, but I don't think that chip_layout is good, but we also declare the control hardware that is connected to the chip, so it is not only about the content of the chip.
What about `make_chip_connectivity_xxx.py`?
I'll create a poll to get people's opinions :)